src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.cpp

```diff
@@ -2,96 +2,143 @@
 {
 }
 
-void BallPlacementPlayFSM::kickOffWall(const Update &event)
+void BallPlacementPlayFSM::alignWall(const Update &event)
 {
-    PriorityTacticVector tactics_to_run = {{}};
+    PriorityTacticVector tactics_to_run = {{align_wall_tactic}};
+
+    Point ball_pos           = event.common.world_ptr->ball().position();
+    Rectangle field_boundary = event.common.world_ptr->field().fieldBoundary();
+
+    pickoff_final_orientation = calculateWallPickOffDest(ball_pos, field_boundary).first;
+
+    pickoff_point =
+        ball_pos - Vector::createFromAngle(pickoff_final_orientation).normalize(0.4);
 
-    // setup move tactics for robots away from ball placing robot
-    setupMoveTactics(event);
+    align_wall_tactic->updateControlParams(
+        pickoff_point, pickoff_final_orientation, TbotsProto::DribblerMode::OFF,
+        TbotsProto::BallCollisionType::AVOID, {AutoChipOrKickMode::OFF, 0},
+        TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT,
+        TbotsProto::ObstacleAvoidanceMode::AGGRESSIVE);
+
+    setupMoveTactics(event, event.common.num_tactics -
+                                static_cast<unsigned int>(tactics_to_run[0].size()));
     tactics_to_run[0].insert(tactics_to_run[0].end(), move_tactics.begin(),
                              move_tactics.end());
 
-    // setup wall kickoff tactic for ball placing robot
-    Point ball_pos            = event.common.world_ptr->ball().position();
-    Rectangle field_lines     = event.common.world_ptr->field().fieldLines();
-    AutoChipOrKick auto_chick = {AutoChipOrKickMode::AUTOKICK,
-                                 WALL_KICKOFF_VELOCITY_M_PER_S};
+    event.common.set_tactics(tactics_to_run);
+}
+
+void BallPlacementPlayFSM::setPickOffDest(const BallPlacementPlayFSM::Update &event)
+{
+    Point ball_pos           = event.common.world_ptr->ball().position();
+    Rectangle field_boundary = event.common.world_ptr->field().fieldBoundary();
+
+    std::pair<Angle, Point> location = calculateWallPickOffDest(ball_pos, field_boundary);
+
+    pickoff_final_orientation = location.first;
+    pickoff_destination =
+        location.second - Vector::createFromAngle(pickoff_final_orientation)
+                              .normalize(BACK_AWAY_FROM_WALL_M);
+}
+
+void BallPlacementPlayFSM::pickOffWall(const BallPlacementPlayFSM::Update &event)
+{
+    PriorityTacticVector tactics_to_run = {{pickoff_wall_tactic}};
+
+    pickoff_wall_tactic->updateControlParams(
+        pickoff_destination, pickoff_final_orientation, true,
+        TbotsProto::MaxAllowedSpeedMode::BALL_PLACEMENT_WALL_DRIBBLE,
+        TbotsProto::MaxAllowedSpeedMode::BALL_PLACEMENT_WALL_DRIBBLE);
 
-    Angle kick_angle = calculateWallKickoffAngle(ball_pos, field_lines);
-    pivot_kick_tactic->updateControlParams(ball_pos, kick_angle, auto_chick);
-    tactics_to_run[0].emplace_back(pivot_kick_tactic);
+    setupMoveTactics(event, event.common.num_tactics -
```

Comment:
needs safety checks that event.common.num_tactics > tactics_to_run[0].size(). Otherwise, we risk underflowing into an unsigned type.